details.php

```diff
@@ -80,4 +82,20 @@
         .userdate($record->modifieddate, get_string('summarydate', 'mod_cpdlogbook')),
 ]);
 
+$fs = get_file_storage();
+if ($files = $fs->get_area_files($context->id, 'mod_cpdlogbook', 'attachments', $record->id)) {
+    foreach ($files as $file) {
+        $fileurl = moodle_url::make_pluginfile_url($file->get_contextid(), $file->get_component(), $file->get_filearea(),
+        $file->get_itemid(), $file->get_filepath(), $file->get_filename());
+        $downloadurl = $fileurl->get_port() ? $fileurl->get_scheme() . '://' . $fileurl->get_host() . $fileurl->get_path() . ':'
```

Comment:
This should not be required, there should be an existing method which builds this url for you

Reply:
Also I'm confused by $fileurl vs $downloadurl, isn't there only a single url?

Reply:
I think it's right now?